cpp/src/arrow/compute/util_internal.cc

```diff
@@ -20,16 +20,29 @@
 #include "arrow/compute/util.h"
 #include "arrow/memory_pool.h"
 
+#ifdef ADDRESS_SANITIZER
```

Comment:
Some platforms don't have ASAN, so wrap the ASAN related code with a flag that is only defined with ASAN.